src/point_one/fusion_engine/messages/measurements.h

```diff
@@ -1261,6 +1261,59 @@ struct P1_ALIGNAS(4) RawHeadingOutput : public MessagePayload {
   float baseline_distance_m = NAN;
 };
 
+/**
+ * @brief A packet of data with an external definition. (@ref
```

Comment:
I'd suggest we be careful with the word "packet" since some people might interpret it as framed data. How about:
> A block of incoming sensor data whose definition depends on the value of @ ref data_type.

And then
> This message ... data. The payload is not guaranteed to be aligned to a specific message boundary, or to contain complete messages.